### PR TITLE
CI: Automatically open an issue if nightly build fails

### LINKED ISSUE #8734: The CI should automatically open an issue in case of failure
We keep having our nightly builds fail for multiple days until someone realizes that they're broken (latest example: https://gitter.im/lampepfl/dotty?at=5e9880a963e7b73a5fdd7770). Can we set up github actions to automatically open an issue when that happen so we can be faster at fixing these things? A quick google search leads me to https://github.com/JasonEtco/create-an-issue.

### PR DESCRIPTION
**Pull request by griggt**
Tests of this in action may be seen at https://github.com/griggt/trial-open-issue-on-ci-failure

Closes #8734

### CODE REVIEW

File: .github/workflows/issue_nightly_failed.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+title: Nightly {{ workflow }} workflow of {{ date | date('YYYY-MM-DD') }} failed
+labels: area:infrastructure
```
Comment (griggt) [inline .github/workflows/issue_nightly_failed.md:3]: Which label(s), if any, should be applied to the opened issue?

Comment (smarter) [reply .github/workflows/issue_nightly_failed.md:3]: i would say itype:bug, prio:blocker

**Review by smarter — APPROVED** [review]
This should be super useful; thanks!

**Comment by SethTisue** [comment]
very cool

**Pull request merged**